docs/topics/custom-images.md

```diff
@@ -0,0 +1,103 @@
+# Custom images
+
+This document will help you get a CAPZ Kubernetes cluster up and running with your custom image.
+
+## Reference images
+
+An *image* defines the operating system and Kubernetes components that will populate the disk of each node in your cluster.
+
+By default, images offered by "capi" in the [Azure Marketplace][azure-marketplace] are used. You can list these *reference images* with this command:
+
+```bash
+az vm image list --publisher cncf-upstream --offer capi --all -o table
+```
+
+But for more control over your nodes, you should build, publish, and use a *custom image*.
+
+## Building a custom image
+
+Cluster API uses the Kubernetes [Image Builder][image-builder] tools. You can use the [Azure image][image-builder-azure] from that project as a starting point for your custom image.
+
+[The Image Builder Book][capi-images] explains how to customize and build the images defined in that repository, with instructions for [Azure CAPI Images][azure-capi-images] in particular.
+
+### Operating system expectations
+
+TODO!
+
+### Required components
+
+TODO!
+
+## Publishing a custom image
+
+TODO!
+
+## Creating a cluster with your custom image
+
+To use a custom image, it needs to be referenced in an `image:` section of your `AzureMachineTemplate`. See below for more specific examples.
+
+### Using Shared Image Gallery
+
+To use an image from the [Shared Image Gallery][shared-image-gallery], fill in the `subscriptionID`, `resourceGroup`, `gallery`, `name`, and `version` fields:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureMachineTemplate
+metadata:
+  name: capz-shared-gallery-example
+spec:
+  template:
+    spec:
+      image:
+        subscriptionID: ...
+        resourceGroup: ...
+        gallery: ...
+        name: ...
+        version: ...
+```
+
+Please also see the [replication recommendations][replication-recommendations] for the Shared Image Gallery.
+
+### Using Azure Marketplace
+
+To use an image from [Azure Marketplace][azure-marketplace], populate the `publisher`, `offer`, `sku`, and `version` fields:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureMachineTemplate
+metadata:
+  name: capz-marketplace-example
+spec:
+  template:
+    spec:
+      image:
+        publisher: ...
+        offer: ...
+        sku: ...
+        version: ...
+```
+
+### Using image ID
+
+To use an image by ID, only the `id` field must be set:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureMachineTemplate
+metadata:
+  name: capz-image-id-example
+spec:
+  template:
+    spec:
+      image:
+        id: ...
```

Comment:
let's put an example fake image ID in here to show an example of what format we expect